validator/src/validator.rs

```diff
@@ -366,7 +407,11 @@ where
                     equivocation_proofs,
                     prev_seed,
                     next_block_number,
-                    Self::PRODUCER_TIMEOUT,
+                    Duration::from_millis(self.compute_micro_block_producer_timeout(
+                        timestamp,
+                        head.timestamp(),
+                        Self::BLOCK_DISTANCE_FROM_HEAD,
```

Comment:
This might need adjustment, because there aren't always `Self::BLOCK_DISTANCE_FROM_HEAD` blocks into the past (near the PoS genesis block, or do we look into PoW history)?

Reply:
Fixed